docs/writing-importers.md

```diff
@@ -0,0 +1,33 @@
+# Writing Importers
+
+Importers in HMCI are classes that extend the `HMCI\Iterator\Base` class. This class provides a standard interface for iterating over data, and a standard interface for processing data. An importer is typically a single data type, so it's likely that a project migration will have multiple importers.
+
+Think of an importer as a "data source" which will fetch items to be imported, and then process each iteam to be imported. An importer need only implement the `get_items( int $offset, int $count ): array<mixed>` method, which is used to fetch items from the data source. The type of each item returned by this method is up to the importer. Optionally `get_count(): int` can be implemented to return the total number of items in the data source, if the count can be calculated in a more performant way than `count( while ( get_items() )... )`.
+
+The importer class then must implement the `process_item( mixed $item ) : bool` method, which is responsible for importing the item. This method is passed each item returned by the `get_items()` method. The bulk of the importer logic is implemented in this method.
+
+## Iterators
+
+As mentioned, all importers should extend the `HMCI\Iterator\Base` class. That is because importers _are_ `Iterators`. It's unlikley you'll extend the `HMCI\Iterator\Base` class directly though, as HMCI providers a number of base iterators that you can extend to create your own iterators. For example, if you are importering data from a collection of CSV file, you can extend the `HMCI\Iterator\Files\CSV` class to create your own iterator. This way you will only need to implement the `process_item( mixed $item ) : bool` method, and HMCI will handle the rest.
```

Comment:
```suggestion
As mentioned, all importers should extend the `HMCI\Iterator\Base` class. That is because importers _are_ `Iterators`. It's unlikely you'll extend the `HMCI\Iterator\Base` class directly though, as HMCI providers a number of base iterators that you can extend to create your own iterators. For example, if you are importing data from a collection of CSV files, you can extend the `HMCI\Iterator\Files\CSV` class to create your own iterator. This way you will only need to implement the `process_item( mixed $item ) : bool` method, and HMCI will handle the rest.
```